msdm/algorithms/lrtdp.py

```diff
@@ -50,16 +51,14 @@ def plan_on(self, mdp: MarkovDecisionProcess):
         res = self.res
         res.policy = AssignmentMap()
         res.Q = AssignmentMap()
-        for s in mdp.state_list:
+        for s in sum(self.res.trials, []):
+            if s in res.policy:
+                continue
             res.policy[s] = AssignmentMap([(self.policy(mdp, s), 1)])
```

Comment:
should remove this (and initialization of `res.policy` above) since it's overwritten below